test/Microsoft.AspNet.Routing.Tests/RouteCollectionTest.cs

```diff
@@ -228,7 +259,78 @@ public void NamedRouteTests_GetNamedRoute_AmbiguousRoutesInCollection_ThrowsForA
 
             // Act & Assert
             var ex = Assert.Throws<InvalidOperationException>(() => routeCollection.GetVirtualPath(virtualPathContext));
-            Assert.Equal("The supplied route name 'ambiguousRoute' is ambiguous and matched more than one route.", ex.Message);
+            Assert.Equal(
+                "The supplied route name 'ambiguousRoute' is ambiguous and matched more than one route.", 
+                ex.Message);
+        }
+
+        // "Integration" tests for RouteCollection
+
+        public static IEnumerable<object[]> DataTokens
```

Comment:
what does this mean? `DataTokens` is a defined concept in routing, and I don't think this is it.